### DISCUSSION TITLE
LO-TO splitting and Born effective charges

### DISCUSSION THREAD

**Discussion by biglinn** [SSCHAcode/python-sscha#344] [Q&A] [answered]
Hi, @mesonepigreco:
I have a doubt about LO-TO splitting.
I check the dyn1 file(gamma point). LO mode and TO mode are degenerate.
![image](https://github.com/SSCHAcode/python-sscha/assets/155951912/5325a0db-67cc-41cd-9b29-c4f2cefeb526)
But LO and TO is splitted after I plot phonon spectrum with the script from tutorial. Main codes are:
```
dyn = CC.Phonons.Phonons("harmonic_dyn", 4)

qpath, data = CC.Methods.get_bandpath(dyn.structure.unit_cell,
                                      PATH,
                                      SPECIAL_POINTS,
                                      N_POINTS)
xaxis, xticks, xlabels = data # Info to plot correclty the x axis

dispersion = CC.ForceTensor.get_phonons_in_qpath(dyn, qpath)
```
I look up the CellConstructor Documentation, but I don't find "Methods.get_bandpath" and "ForceTensor.get_phonons_in_qpath".
Sometimes, phonon spectrum doesn't show LO-TO splitting if the dyn matrices are not very good.
So, I wonder that how the plot script leads to LO-TO splitting.

**Comment by biglinn** [comment]
Moreover, if I use `dyn = dyn.Interpolate( (2,2,2), (3,3,3))` to give supercell, Born effective charges are ignored.
I find a code `InterpolateMesh` to replace `Interpolate` in the document.
![image](https://github.com/SSCHAcode/python-sscha/assets/155951912/75566cb3-6457-4497-b892-505164530924)

However, there is a problem.
![image](https://github.com/SSCHAcode/python-sscha/assets/155951912/057ccbbc-836b-4802-98af-c89f56069668)

It seems that this parameter `lo_to_splitting` is not defined in this function.
I try to run `dyn = dyn.InterpolateMesh((3,3,3), lo_to_splitting = True)`, but there are still not Born charges in dyn files.
So how to generate Born effective charges and open LO-TO splitting?

**Reply by mesonepigreco** [reply]
Do not use InterpolateMesh as it is still experimental and may lead to incorrect results.

**Comment by mesonepigreco** [comment, marked as answer]
The code automatically looks for Born Effective charges inside the first dynamical matrix (in quantum espresso format). 
If you computed it with quantum espresso ph.x, with epsil = .true., then you should have the effective charges stored inside the dynamical matrix. Cellconstructor reads automatically those values. When interpolating the dispersion using the ForceTensor module, as  done in:

```
dispersion = CC.ForceTensor.get_phonons_in_qpath(dyn, qpath)
```
The code automatically takes the effective charges and computes the LO-TO splitting for the given path.
However, at Gamma, LO-TO splitting is not well defined as it is nonanalytical (the eigenvector of the LO-TO modes depends on the direction in which you approach to Gamma). This is why it is not reported in the decomposition of eigenvalues and eigenvectors.

As you noticed, the Interpolate function does not exploit LO-TO splitting. This should be fixed as you noticed by InterpolateMesh, that employs the ForceTensor module as the ``get_phonons_in_qpath``, which uses the info of the effective charges to calculate LO-TO. However, the InterpolateMesh method has still some bugs and may lead to wrong results, I do not recommend using it.

**Reply by biglinn** [reply]
Indeed, is all longrange information about LO-TO splitting stored in Born effective charges?
`Interpolate` ingnores effective charges. So could I use dyn files without effective charges to do SSCHA minimization, and manually add effective charges to the dyn files for plotting phonon spectrum with `ForceTensor.get_phonons_in_qpath`?

**Reply by biglinn** [reply]
`Interpolate` is used to generate supercell of different size. So I only it before first SSCHA minimization.
If I want do SSCHA minimization with effective charges, could I do as following:

After using `Interpolate` to generate dyn_start_population1_XX, I manually add effective charges to the dyn files to do SSCHA minimization.

Does this two ways(with/without effective charges)  have some difference?

**Reply by mesonepigreco** [reply]
Hi,

The SSCHA minimization ignores the effective charges, as LO-TO only matters during interpolation. When plotting the phonon band structure, you are interpolating, so they should be added to the dynamical matrix. 

If you use Interpolate to get the starting dynamical matrix of the SSCHA, you will have a "worse" starting guess, as LO-TO will not be included in that. However, that is not a real issue, as the SSCHA minimization will automatically account for long-range interactions when computing the forces from the ab initio calculation (maybe it will take a few more iterations to converge). 

However, why are you using Interpolate to get the starting SSCHA dynamical matrix and not directly computing the harmonic dynamical matrix in the supercell you want? This way, you will have the correct LO-TO for all q-points from the beginning. Usually, the harmonic calculation is cheaper than the SSCHA in the same q-mesh.

**Reply by biglinn** [reply]
Thanks! I get it.

Maybe I'm obsessed with it. 
I want do calcutions in different supercell, such as 222, 333, 444. So I calculate dynmical matrices with 222 q-mesh by QE and use Interpolate( (2,2,2), (X,X,X) ) to get corresponding dynmical matrices and scf files.

**Reply by biglinn** [reply]
Hi, @mesonepigreco:
I manually add effective charges to the dyn files for plotting phonon spectrum. But there is a little imaginary phonon near Gamma point.
Does this have something to do with ASR rule? Or maybe the parameters nk1, nk2, nk3 of ph.x calculations? 
![population40](https://github.com/user-attachments/assets/48d03f77-3041-46fb-9c6f-cbb423f98663)

**Reply by mesonepigreco** [reply]
Hi, it may be related to many things, most likely the mesh of q points for the direct calculations. I believe there are advanced ASR implementations to cure these things in modern QE versions (but I do not know how to use them), or it could be a slightly non perfectly relaxed cell or phonon calculation.

**Reply by biglinn** [reply]
It's so complex. I'll make tests. 
Thanks!

**Reply by biglinn** [reply]
Hi, @mesonepigreco:
I notice that the first tutorial use a method:
```
relax = sscha.Relax.SSCHA(minim, calculator, N_configs = N_CONFIGS, max_pop = MAX_ITERATIONS)
……
# Run the NVT simulation
relax.relax(get_stress = True)
```
This `relax.relax` call the function `compute_ensemble` to calculate energies and forces from ensembles with ASE calculators. 
And if I'm likely to use NPT ensembles, I can use `relax.vc_relax`.

So, could I use this method with DFT instead of ASE? 
Since the second tutorial "Setting up automatic SSCHA simulations on clusters" does not use `sscha.Relax.SSCHA` method,  it just use `ensemble.generate` and `ensemble.save` to get configurations and do minimization.

**Reply by biglinn** [reply]
Oh, sorry! The second tutorial has introduced `relax` method in the back part, I'll study it.
